src/celeritas/em/interactor/WentzelInteractor.hh

```diff
@@ -0,0 +1,159 @@
+ * scattering off of nuclei and atomic electrons. Scattering off of the nucleus
+ * versus electrons is randomly sampled based on the relative cross-sections.
+ * No secondaries are created in this process (in the future, with hadronic
+ * transport support, secondary ions may be emitted), however production cuts
+ * are used to determine the maximum scattering angle off of electrons.
+ *
+ * \note This performs the same sampling as in Geant4's
+ *  G4eCoulombScatteringModel, as documented in section 8.2 of the Geant4
+ *  Physics Reference Manual (release 11.1).
+ */
+class WentzelInteractor
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    using MomentumSq = units::MevMomentumSq;
+    //!@}
+
+  public:
+    //! Construct with shared and state data
+    inline CELER_FUNCTION WentzelInteractor(WentzelRef const& shared,
+                                            ParticleTrackView const& particle,
+                                            Real3 const& inc_direction,
+                                            IsotopeView const& target,
+                                            ElementId const& el_id,
+                                            CutoffView const& cutoffs);
+
+    //! Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Incident direction
+    Real3 const& inc_direction_;
+
+    // Incident particle
+    ParticleTrackView const& particle_;
+
+    // Target isotope
+    IsotopeView const& target_;
+
+    // Scattering direction distribution of the Wentzel model
+    WentzelDistribution const sample_direction;
+
+    //// HELPER FUNCTIONS ////
+
+    //! Calculates the recoil energy for the given scattering direction
+    inline CELER_FUNCTION real_type calc_recoil_energy(
+        Real3 const& new_direction, Mass const& target_mass) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from shared and state data
+ */
+CELER_FUNCTION
+WentzelInteractor::WentzelInteractor(WentzelRef const& shared,
+                                     ParticleTrackView const& particle,
+                                     Real3 const& inc_direction,
+                                     IsotopeView const& target,
+                                     ElementId const& el_id,
+                                     CutoffView const& cutoffs)
+    : inc_direction_(inc_direction)
+    , particle_(particle)
+    , target_(target)
+    , sample_direction(particle,
+                       target,
+                       shared.elem_data[el_id],
+                       value_as<Energy>(cutoffs.energy(particle.particle_id())),
```

Comment:
Should this be the secondary cutoff energy for protons rather than the cutoff for the incident electron or positron? (see [here](https://github.com/Geant4/geant4/blob/dd1f179cda58f54140945ad67846ff417903a862/source/processes/electromagnetic/standard/src/G4CoulombScattering.cc#L67) and [here](https://github.com/Geant4/geant4/blob/dd1f179cda58f54140945ad67846ff417903a862/source/processes/electromagnetic/standard/src/G4eCoulombScatteringModel.cc#L140))

Reply:
This confusion is partly because the Coulomb process is not added yet. I think that @amandalund pointed the right places (i.e, setting the secondary == proton) so that the associated cut should be the proton cut, which should be applied to all places of the model no matter what the incident particle (i.e, any below 1 GeV) is.  Here, only the e-(e+) and the nucleus scattering is considered along with NIEL while the e-e-(e+/e-) scattering is covered by the Moller(Bhabha) scattering and the usual dEdx.  Another TODO is that we need to add the generic ion production when we add the lepton-nuclear in the future - I explicitly asked @hhollenb to skip the part for now as we do not want to deal with secondary particles other than e+/e-/gamma.

Reply:
It looks like [`CutoffParams`](https://github.com/celeritas-project/celeritas/blob/ddf0d65a91c4bf16433d167ef0bf1e7f147f91f0/src/celeritas/phys/CutoffParams.cc#L140) doesn't support protons. The error I'm getting is that protons don't have valid indexes in the cutoff table, which is filled by `pdg_numbers()` [here](https://github.com/celeritas-project/celeritas/blob/ddf0d65a91c4bf16433d167ef0bf1e7f147f91f0/src/celeritas/phys/CutoffParams.cc#L99). Shall I go ahead and add protons to supported particles in `CutoffParams`?

Reply:
We usually don't read proton data (using the Geant "import" options) since we don't support proton transport. Maybe this can also be skipped?

Reply:
@stognini Are we supporting (i.e., importing) the cutoff values for protons?

Reply:
Okay, why don't see leave the code as it is with a note (i.e, for changing the cut to proton cutoff when it is supported).